packages/@sanity/schema/src/sanity/extractSchema.ts

```diff
@@ -210,7 +210,7 @@ export function extractSchema(
       attributes[field.name] = {
         type: 'objectAttribute',
         value,
-        optional: extractOptions.enforceRequiredFields ? fieldIsRequired : true,
+        optional: extractOptions.enforceRequiredFields ? fieldIsRequired === false : true,
```

Comment:
I feel this is more readable ⬇️  Alternatively we can add a comment above this line. 
```suggestion
        optional: !extractOptions.enforceRequiredFields || !fieldIsRequired
```

Reply:
I do disagree 🙈 ternarys makes it more explicit imo. Can move it out and add a comment though

Reply:
a comment sounds good to me 🤝